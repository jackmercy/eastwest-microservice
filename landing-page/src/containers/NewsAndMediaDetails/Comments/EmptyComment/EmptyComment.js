import { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { useIntl } from 'react-intl';

import { Button } from '../../../../components';
import { scrollToElement } from '../../../../utils';

import useStyles from './emptyComment.styles';

const EmptyComment = ({ className }) => {
  const intl = useIntl();
  const classes = useStyles();

  const handleClick = useCallback(() => {
    scrollToElement('main-comment-box', { offset: 20 });
    setTimeout(() => {
      const el = document.querySelector('#main-comment-box textarea');
      if (el) {
        el.focus();
      }
    }, 500);
  }, []);

  return (
    <div className={classNames(classes.root, className)}>
      <img
        alt="empty"
        src="/static/images/news/empty-comment.png"
        className={classes.image}
        data-test="empty-comment-image"
      />
      <h2 className={classes.title} data-test="empty-comment-title">
        {intl.formatMessage({ id: 'NEWS.DETAILS.COMMENT.EMPTY.TITLE' })}
      </h2>
      <p className={classes.description} data-test="empty-comment-description">
        {intl.formatMessage({ id: 'NEWS.DETAILS.COMMENT.EMPTY.DESCRIPTION' })}
      </p>
      <Button
        color="primary"
        className={classes.button}
        dataTest="empty-comment-button"
        onClick={handleClick}
      >
        {intl.formatMessage({ id: 'NEWS.DETAILS.COMMENT.EMPTY.BUTTON' })}
      </Button>
    </div>
  );
};
EmptyComment.propTypes = {
  className: PropTypes.string,
};
EmptyComment.defaultProps = {
  className: '',
};

export default memo(EmptyComment);
