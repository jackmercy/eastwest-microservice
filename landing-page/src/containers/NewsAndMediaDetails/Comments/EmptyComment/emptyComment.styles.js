import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    padding: [rem(50), rem(16)],
    textAlign: 'center',
  },
  image: {
    display: 'block',
    width: '100%',
    maxWidth: rem(570),
    margin: [0, 'auto'],
  },
  title: {
    fontWeight: 'bold',
    fontSize: rem(32),
    lineHeight: rem(48),
    marginTop: rem(32),
    marginBottom: 0,
  },
  description: {
    fontSize: rem(16),
    lineHeight: rem(24),
    margin: 0,
  },
  button: { marginTop: rem(32), minWidth: rem(145) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: {},
    image: {},
    title: {},
    description: {},
    button: {},
  },
});
