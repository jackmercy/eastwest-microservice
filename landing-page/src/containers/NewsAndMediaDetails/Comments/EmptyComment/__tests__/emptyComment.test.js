import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import EmptyComment from '../EmptyComment';

describe('Testing NewsAndMediaDetails > Comments > EmptyComment', () => {
  const props = {
    paths: [
      {
        label: 'News and media',
        href: '/news-and-media',
      },
      {
        label: 'Technology news',
        href: '/news-and-media',
      },
      {
        label:
          // eslint-disable-next-line max-len
          'iPhone SE has some of the best iPhone 11 features, including wireless charging',
      },
    ],
  };
  it('should render match snapshot', () => {
    const wrapper = shallow(<EmptyComment {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
