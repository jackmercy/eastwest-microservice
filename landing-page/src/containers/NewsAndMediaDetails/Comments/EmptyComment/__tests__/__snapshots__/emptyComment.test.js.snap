// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsAndMediaDetails > Comments > EmptyComment should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <img
    alt="empty"
    className="image-0-2-2"
    data-test="empty-comment-image"
    src="/static/images/news/empty-comment.png"
  />
  <h2
    className="title-0-2-3"
    data-test="empty-comment-title"
  >
    No comment
  </h2>
  <p
    className="description-0-2-4"
    data-test="empty-comment-description"
  >
    Become the first person to comment on this article
  </p>
  <Memo([object Object])
    className="button-0-2-5"
    color="primary"
    dataTest="empty-comment-button"
    onClick={[Function]}
  >
    Leave comment
  </Memo([object Object])>
</div>
`;
